solver/src/driver.rs

```diff
@@ -285,23 +286,58 @@ impl Driver {
         }
     }
 
-    // Record metric with the amount of orders that were matched but not settled in this runloop (effectively queued for the next one)
-    // Should help us to identify how much we can save by parallelizing execution.
-    fn report_matched_but_unsettled_orders(
+    /// Record metrics on the matched orders from a single batch. Specifically we report on
+    /// the number of orders that were;
+    ///  - matched but not settled in this runloop (effectively queued for the next one)
+    ///  - matched but cancelled
+    ///  - matched but coming from known liquidity providers.
+    /// Should help us to identify how much we can save by parallelizing execution.
+    fn report_matched_orders(
         &self,
         submitted: &Settlement,
         all: impl Iterator<Item = RatedSettlement>,
+        orders: Vec<LimitOrder>,
     ) {
         let submitted: HashSet<_> = submitted
             .trades()
             .iter()
             .map(|trade| trade.order.order_meta_data.uid)
             .collect();
-        let all_matched: HashSet<_> = all
+        let all_matched_orders: HashSet<_> = all
             .flat_map(|solution| solution.settlement.trades().to_vec())
-            .map(|trade| trade.order.order_meta_data.uid)
+            .map(|trade| trade.order)
+            .collect();
+        let all_matched_ids: HashSet<_> = all_matched_orders
+            .iter()
+            .map(|order| order.order_meta_data.uid)
             .collect();
-        let matched_but_not_settled: HashSet<_> = all_matched.difference(&submitted).collect();
+        let matched_but_not_settled = all_matched_ids.difference(&submitted).copied().collect();
+        let cancelled_order_ids: HashSet<_> = all_matched_orders
+            .iter()
+            .filter_map(|order| match order.order_meta_data.invalidated {
+                true => Some(order.order_meta_data.uid),
+                false => None,
```

Comment:
Can this ever be true? If I understand the code right we're passing in *sovlable* orders. Those shouldn't be invalidated. Thinking this further I wonder if the driver is even the right place to collect these metrics. We have all the data in Postgres so we could write a script (or rust program) that at any time can create these metrics retroactively instead of doing them live through Prometheus like this.

Reply:
Yea, perhaps we should just leave canceled orders out for this PR and focus on excluding the liquidity provider orders here. Your suggestion might be worth doing separately. but the purpose of this PR is really meant to focus on our grafan metric not showing bad settlement matching results (i.e. right now its showing < 40% matched) because of a bunch of spam liquidity provider orders.